src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -140,6 +140,24 @@ internal static void ValidateScriptCompilationParameters(Compilation previousScr
             }
         }
 
+        internal static void CheckCompilationOptions(CompilationOptions options)
+        {
+            if (options == null)
+            {
+                return;
+            }
+
+            if (options.PublicSign)
+            {
+                var keyFile = options.CryptoKeyFile;
+                if (!string.IsNullOrWhiteSpace(options.CryptoKeyFile) &&
```

Comment:
I wouldn't allow whitespace. Only null should be used to indicate the path is missing.